cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatabaseImpl.java

```diff
@@ -217,8 +209,7 @@ protected DatabaseImpl(String dir, String name, KeyProvider provider) throws SQL
 
     @Override
     public String getDatastoreName() {
-        Misc.checkState(this.isOpen(), "Database is closed");
-        return this.datastoreName;
+        return this.datastoreDir.getName();
```

Comment:
I wonder if we should return something longer here. Per the javadoc
>This is just the last name in the pathname's name sequence.

Now that we don't have the `DatastoreManager` context we have lost the filesystem's enforced uniqueness of a an individual datastore name (folder) within that context.

I wonder if this new name method should return the whole path?